File: src/providers/nvidia.ts
import { createOpenAICompatible } from "@ai-sdk/openai-compatible";
import type { ProviderAdapter } from "./base.js";
import { baseURLFrom } from "./base.js";

export const nvidiaProvider: ProviderAdapter = {
  name: "NVIDIA",
  source: "nvidia",
  apiKeyEnvVar: "NVIDIA_API_KEY",
  messageProtocol: "openai",
  chatModel(modelId: string, apiKey: string, customFetch?: typeof fetch) {
    const provider = createOpenAICompatible({
      name: "nvidia",
      baseURL: baseURLFrom(
        "SLEEPYROUTER_NVIDIA_BASE_URL",
        "https://integrate.api.nvidia.com/v1",
      ),
      apiKey,
      fetch: customFetch,
    });
    return provider.chatModel(modelId);
  },
};

File: sleepyrouter/providers/nvidia.py
"""NVIDIA provider adapter."""

from .base import BaseProviderAdapter


class NVIDIAProviderAdapter(BaseProviderAdapter):
    def __init__(self) -> None:
        super().__init__(
            name="NVIDIA",
            source="nvidia",
            api_key_env_var="NVIDIA_API_KEY",
            api_base="https://integrate.api.nvidia.com/v1",
        )
